cpp/src/arrow/type.cc

```diff
@@ -3294,7 +3318,7 @@ void InitStaticData() {
   // * Time32
   // * Time64
   // * Timestamp
-  g_primitive_types = {null(), boolean(), date32(), date64()};
+  g_primitive_types = {null(), boolean(), date32(), date64(), binary_view(), utf8_view()};
```

Comment:
The view types probably shouldn't be part of base binary since they do not inherit `BaseBinaryType`.

`g_primitive_types` contains all of the base binary types too: 

https://github.com/bkietz/arrow/blob/6f2e851663cf8d0345796c415b56de182f260ed3/cpp/src/arrow/type.cc#L3321-L3323

If it'd be useful to have a new category like `BinaryViewTypes()` that could be a follow up.